packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -252,6 +253,12 @@ export const useGridColumnResize = (
           `[data-field="${colDef.field}"]`,
         )!;
 
+      colGroupingElementRef.current = Array.from(
+        apiRef.current.columnHeadersContainerElementRef?.current!.querySelectorAll<HTMLDivElement>(
+          `[data-fields~="${colDef.field}"]`,
```

Comment:
Might not be very reliable when `field` attributes contain space, for example:

https://codesandbox.io/s/basicgroupingdemo-demo-mui-x-forked-s2b8ts?file=/demo.tsx

Try resizing `id` column

Reply:
As soon as a `field` is a sub-string of another one the bug will occurs

A solution could be to use custom delimiter as follow

```jsx
// In the group header
data-fields={`-${fields.join('-')}-`}

// In the selector
`[data-fields~="-${colDef.field}-"]`
```

With that the two column fields should be `id` and `id-2` (Ok, that's possible too. Maybe with a more special character)

Reply:
Yeah, the delimiter with special characters should be enough to lower the risk of this issue :+1: